server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2627,7 +2627,7 @@ public boolean postStateTransitionEvent(final StateMachine2.Transition<State, Vi
                 s_logger.info("Schedule a router reboot task as router " + vo.getId() + " is powered-on out-of-band. we need to reboot to refresh network rules");
                 _rebootRouterExecutor.execute(new RebootTask(vo.getId()));
         } else {
-            if (isOutOfBandMigrated(opaque)) {
+            if (isOutOfBandMigrated(opaque) && (vo.getType() == VirtualMachine.Type.DomainRouter)) {
```

Comment:
@koushik-das A first shot:

        final VirtualMachine.Event event = transition.getEvent();
        boolean outOfBandMigrated = isOutOfBandMigrated(opaque);
        boolean reprovision_out_of_band = RouterReprovisionOnOutOfBandMigration.value() && outOfBandMigrated;
        if ( (vo.getType() == VirtualMachine.Type.DomainRouter)) {
            if (
                    (event == VirtualMachine.Event.FollowAgentPowerOnReport) &&
                    (newState == State.Running) &&
                    ((oldState == State.Stopped) || (reprovision_out_of_band))
                ) {
                s_logger.info("Schedule a router reboot task as router " + vo.getId() + " is powered-on out-of-band. we need to reboot to refresh network rules");
                _rebootRouterExecutor.execute(new RebootTask(vo.getId()));
            } else {
                if (outOfBandMigrated) {
                    final String title = "Router has been migrated out of band: " + vo.getInstanceName();
                    final String context =
                            "An out of band migration of router " + vo.getInstanceName() + "(" + vo.getUuid() + ") was detected. No automated action was performed.";
                    _alertMgr.sendAlert(AlertManager.AlertType.ALERT_TYPE_DOMAIN_ROUTER, vo.getDataCenterId(), vo.getPodIdToDeployIn(), title, context);
                }
            }
        }
It can probably be more clean up even. I saw none but dummy tests are there for this class so I want to create that as well before committing